### PR TITLE
Terms and conditions

### PR DESCRIPTION
**Pull request by okbel**
## What does this PR do?
- Adds the ability to dispatch `enableSubmitSignUpForm` and `disableSubmitSignUpForm` for `talk-plugin-auth`
- Adds `talk-plugin-health-report` that adds one last field to the Sign Up Form

More here: https://www.pivotaltracker.com/n/projects/1863625

Note: The translation service does not support receiving elements I was going to add an update to it but it's not worth it since we are going to upgrade our translation service. And more changes to it 

`talk-plugin-health-report` will be removed from this PR as soon as the review is done :)

### CODE REVIEW

**Comment by wyattjoh** [comment]
`talk-plugin-health-report` should be extracted to another repository @okbel, as it's not a part of the core installation.

**Comment by okbel** [comment]
👍 I'll leave it for testing and then I'll remove that folder. The PR is needed though, it has updates to the `talk-plugin-auth`

**Comment by okbel** [comment]
@cvle ready to be checked! Thanks for the `hasBlocker` pattern 😄

File: plugins/talk-plugin-auth-checkbox/client/components/TermsAndConditionsField.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Checkbox } from 'plugin-api/beta/client/components/ui';
+import { t } from 'plugin-api/beta/client/services';
+import styles from './TermsAndConditionsField.css';
+import cn from 'classnames';
+
+const TermsLink = () => (
+  <a
+    className={styles.link}
+    href={t('talk-plugin-health-report.terms-link')}
```
Comment (alinhle) [inline plugins/talk-plugin-auth-checkbox/client/components/TermsAndConditionsField.js]: Is the name `talk-plugin-health-report` here correct while the plugin name is `talk-plugin-auth-checkbox`?

File: plugins/talk-plugin-auth-checkbox/client/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import TermsAndConditionsField from './components/TermsAndConditionsField';
+import translations from './translations.yml';
+
+export default {
+  slots: {
+    'talkPluginAuth-formField': [TermsAndConditionsField],
```
Comment (alinhle) [inline plugins/talk-plugin-auth-checkbox/client/index.js]: It feels a bit odd using the dash in combination with camelCase. What do you think of `talkPluginAuth.formField` instead?

**Review by alinhle — CHANGES_REQUESTED** [review]
I removed `docs/source/_data/introspection.json` from the PR which seems to have been accidently commited and added it to `.gitignore`. I also changed the plugins service to produce a descriptive key which I replaced throughout the codebase where we relied on the talkPluginName instead. This makes sure no matter how many components a plugins adds to a slot, it still works. 

I adapted your blockers API to the above changed and simplified it a little bit. Besides the small review comments, it looks good!

File: plugins/talk-plugin-auth-checkbox/client/index.js (hunk 2 above)
Comment (okbel) [reply plugins/talk-plugin-auth-checkbox/client/index.js]: I like it :D

**Review by alinhle — DISMISSED** [review]

**Review by kgardnr — APPROVED** [review]

**Pull request merged**